docs/src/modules/components/AppDrawerNavItem.js

```diff
@@ -65,42 +92,39 @@ export default function AppDrawerNavItem(props) {
 
   if (href) {
     return (
-      <ListItem className={classes.itemLeaf} disableGutters {...other}>
-        <Button
-          color="inherit"
-          component={Link}
+      <li className={classes.liLeaf} {...other}>
```

Comment:
This can now use `ListItem` again, no?

Reply:
I don't understand the dogfooding argument on this one.

I don't see the problems using the `ListItem` would help solve the drawer nav links. ListItem renders a `<li>`. We can do that directly. `ListItem` applies CSS we don't need (It's simpler to add custom styles, and start from a rower baseline).

If we wanted to globally customize the drawer nav links, we could be interested in using a custom theme object. In which case, we would still want to have list items and side nav items render differently. Hence an argument for pushing #15107. I think that it's also an argument for rewriting MenuItem to not be built on top of ListItem.

Reply:
> ListItem applies CSS we don't need (It's simpler to add custom styles, and start from a rower baseline).

What are the styles you're adding?

So you would say that with Material design you wouldn't use a `ListItem` in a `Drawer`. Sounds like we should publish a `DrawerListItem` component then.

Reply:
> What are the styles you're adding?

My incentive is to start from the browser defaults, knowing they will almost never change than the style of the ListItem that might and that will give a head heck to handle later on. It's also about the ListItem no providing value for the use case (AFAIK).

> So you would say that with Material design you wouldn't use a ListItem in a Drawer. Sounds like we should publish a DrawerListItem component then.

Yes, definitely. Google uses different styles on its product for the two use cases (list for navigation vs list for the main layout).

Reply:
So now Material design is used as an argument again. Interesting. Will track this in an issue.

Reply:
I would summarize the argument as: in most design systems, drawer navigation item and list item have a different look, hence different components. It seems that the only logic we could share is the reset of the padding, list-style, and margin. I doubt its worth an abstraction.